### ISSUE TITLE
CopyQ 9.0.0 crashes silently and intermittently on macOS (M1)

### ISSUE THREAD

**Issue by lkrms** [hluk/CopyQ#2846]
**Describe the bug**

A bit of a frustrating one because I don't notice anything has happened until I need to pull up my clipboard history... only to find CopyQ hasn't been running for hours/days 😬

Sufficiently intermittent that I haven't been able to replicate it consistently or correlate it with anything else happening on my system.

The only trail of evidence I can find is a trio of entries in `copyq.log` from each disappearance:

```log
CopyQ ERROR [2024-09-17 13:29:54.249] <Client-46026>: Connection lost!
CopyQ ERROR [2024-09-17 13:29:54.249] <monitorClipboard-32684>: Connection lost!
CopyQ ERROR [2024-09-17 13:29:54.249] <Client-32685>: Connection lost!
```

And a macOS crash report:

[CopyQ-2024-09-17-133000.log](https://github.com/user-attachments/files/17024983/CopyQ-2024-09-17-133000.log)

**Version, OS and Environment**

```
CopyQ Clipboard Manager 9.0.0-gdc155ef9-master
Qt: 6.7.0
KNotifications: 5.249.0
Compiler: GCC
Arch: arm64-little_endian-lp64
OS: macOS 14.6
```

**Additional context**

Unlikely to be significant, but these warnings appear in `copyq.log` sporadically, presumably when I'm using certain parts of the UI. They don't coincide with anything going observably wrong:

```log
CopyQ Warning [2024-09-16 15:35:26.981] <Server-30752>: [default] QtWarning: QSystemTrayIcon::setVisible: No Icon set
CopyQ Warning [2024-09-16 15:38:35.881] <Server-30752>: [qt.qpa.fonts] QtWarning: Populating font family aliases took 101 ms. Replace uses of missing font family "Monospace" with one that exists to avoid this cost.
CopyQ Warning [2024-09-16 15:44:31.280] <Server-32198>: [default] QtWarning: QSystemTrayIcon::setVisible: No Icon set
CopyQ Warning [2024-09-16 15:47:59.523] <Server-32198>: [qt.accessibility.table] QtWarning: Cell requested for row 0 is out of bounds for table with 0 rows! Resizing table model.
```

Thanks so much for this excellent piece of software. It's unfortunate to be noticing how much I rely on it in this particular way 😂

**Comment by lkrms** [comment]
> looks as the same as here: #2652

I don't think so, but I'd be happy to be wrong! Once I've signed the `.app` as per that issue, CopyQ runs without any apparent problems for hours/days, i.e. there's no crash on startup, so I don't think it's the same problem. Unless I'm missing something?

**Comment by verydevuser** [comment]
Having the same issue, signing didn't help

**Comment by MrJarnould** [comment]
I also have the same issue. Has anyone found a solution?

**Comment by QNoir** [comment]
> I also have the same issue. Has anyone found a solution?

maybe a trick in [this thread](https://github.com/hluk/CopyQ/issues/2652) helps you.
